### PR TITLE
Hard: withdraws limited by ltv

### PR DESCRIPTION
**Pull request by denalimarsh**
This PR prevents withdraws that would put a user's LTV outside the valid range.

### CODE REVIEW

File: x/harvest/keeper/deposit.go [hunk 1]
```diff
@@ -91,11 +91,39 @@ func (k Keeper) Withdraw(ctx sdk.Context, depositor sdk.AccAddress, coins sdk.Co
...
+	valid, err := k.IsWithinValidLtvRange(ctx, proposedDeposit, borrow)
+	if err != nil {
+		return err
+	}
+
+	if !valid {
+		return sdkerrors.Wrapf(types.ErrInvalidWithdrawAmount, "proposed withdraw outside loan-to-value range")
+	}
+
+	if !deposit.Amount.IsAllGTE(coins) {
```
Comment (karzak) [inline x/harvest/keeper/deposit.go:122]: This seems redundant to the check on line 107.

File: x/harvest/keeper/liquidation.go [hunk 2]
```diff
@@ -276,6 +244,55 @@ func (k Keeper) StartAuctions(ctx sdk.Context, borrower sdk.AccAddress, borrows,
...
+	}
+	if shouldInsertIndex {
+		k.InsertIntoLtvIndex(ctx, currLtv, borrower)
+	}
+	return nil
+}
+
 // GetStoreLTV calculates the user's current LTV based on their deposits/borrows in the store
 // and does not include any outsanding interest.
 func (k Keeper) GetStoreLTV(ctx sdk.Context, addr sdk.AccAddress) (sdk.Dec, bool, error) {
```
Comment (karzak) [inline x/harvest/keeper/liquidation.go:298]: Update function comment to describe what the boolean return variable signifies.

**Review by karzak — APPROVED** [review]
LGTM after nits addressed.

File: x/harvest/keeper/liquidation.go (hunk 2 above)
Comment (denalimarsh) [reply x/harvest/keeper/liquidation.go:298]: https://github.com/Kava-Labs/kava/pull/747/commits/4d78ea83c3a0b2ba4ecafb0087d1c07a364a5f81

File: x/harvest/keeper/deposit.go (hunk 1 above)
Comment (denalimarsh) [reply x/harvest/keeper/deposit.go]: https://github.com/Kava-Labs/kava/pull/747/commits/4d78ea83c3a0b2ba4ecafb0087d1c07a364a5f81

**Pull request merged**